src/main/java/net/dv8tion/jda/internal/handle/GuildRoleUpdateHandler.java

```diff
@@ -121,6 +124,16 @@ protected Long handleInternally(DataObject content)
                             getJDA(), responseNumber,
                             role, wasMentionable));
         }
+
+        RoleIcon oldIcon = role.getIcon();
+        if (!Objects.equals(oldIcon.getIconId(), iconId) || !Objects.equals(oldIcon.getEmoji(), emoji))
```

Comment:
How about `RoleIcon#equals` and `RoleIcon#hashCode`?

Reply:
I can create those, sure, but not sure how those would be helpful here as I don't have the new object before and creating a new one just for equality purposes is pointless